Author: Javi Merino <vicho@debian.org>
Description: Don't run test-gpg if not in a working directory
 test-gpg has to be run in a mercurial working directory as it uses
 that to verify that it hasn't modified the trustdb.gpg file.  Skip
 the test if it is running in an exploded tarball.
Forwarded: http://www.selenic.com/pipermail/mercurial-devel/2012-April/038941.html
--- a/tests/test-gpg.t
+++ b/tests/test-gpg.t
@@ -1,6 +1,10 @@
 Test the GPG extension
 
   $ "$TESTDIR/hghave" gpg || exit 80
+  $ if ! hg identify -q > /dev/null; then
+  >     echo "skipped: not a Mercurial working dir" >&2
+  >     exit 80
+  > fi
   $ cat <<EOF >> $HGRCPATH
   > [extensions]
   > gpg=
